internal/manifest/coreos_installer.go

```diff
@@ -0,0 +1,161 @@
+package manifest
+
+import (
+	"fmt"
+
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/osbuild"
+	"github.com/osbuild/osbuild-composer/internal/platform"
+	"github.com/osbuild/osbuild-composer/internal/rpmmd"
+)
+
+type CoreOSInstaller struct {
+	Base
+
+	// Packages to install in addition to the ones required by the
+	// pipeline.
+	ExtraPackages []string
+
+	// Extra repositories to install packages from
+	ExtraRepos []rpmmd.RepoConfig
+
+	platform     platform.Platform
+	repos        []rpmmd.RepoConfig
+	packageSpecs []rpmmd.PackageSpec
+	kernelName   string
+	kernelVer    string
+	product      string
+	version      string
+	Variant      string
+
+	// Biosdevname indicates whether or not biosdevname should be used to
+	// name network devices when booting the installer. This may affect
+	// the naming of network devices on the target system.
+	Biosdevname bool
+
+	FDO *blueprint.FDOCustomization
+}
+
+// NewCOI creates an CoreOS installer pipeline object.
+func NewCOI(m *Manifest,
+	buildPipeline *Build,
+	platform platform.Platform,
+	repos []rpmmd.RepoConfig,
+	kernelName,
+	product,
+	version string) *CoreOSInstaller {
+	p := &CoreOSInstaller{
+		Base:       NewBase(m, "coi-tree", buildPipeline),
+		platform:   platform,
+		repos:      repos,
+		kernelName: kernelName,
+		product:    product,
+		version:    version,
+	}
+	buildPipeline.addDependent(p)
+	m.addPipeline(p)
+	return p
+}
+
+// TODO: refactor - what is required to boot and what to build, and
+// do they all belong in this pipeline?
+func (p *CoreOSInstaller) coiBootPackageSet() []string {
```

Comment:
I guess I'm picking at naming today :) I think 'getBootPackageSer' would be clearer. coi seems redundant since it's attached to the CoreOSInstaller struct.

Reply:
Thanks for picking the naming nits :)
These are good.